### PR TITLE
feat(Listing) - Listing Module Schema and REST Interfaces

### LINKED ISSUE #20: [Task] Listing Module Schema Update and Interfaces

### PR DESCRIPTION
**Pull request by dlalaj**
Fixes #20 

## Description of the change:
- Update Listing Schema
- Adds GET, POST, PUT, DELETE for the Listing module

## How to manually test:

1. Start dev server by `npm start`
2. Ensure you have created some users to be able to test any of the following routes out.

- `GET /api/listings/:listingId` gets a listing through its ID

```bash
curl --request GET --header 'Content-Type: application/json' localhost:3000/api/listings/:listingId
```

- `GET /api/listings/user/:userId` gets all listings under a given user - the ones this user created

```bash
curl --request GET --header 'Content-Type: application/json' localhost:3000/api/listings/user/:userId
```

- `POST /api/listings` create listing for a given user - here listerID on the body should be provided by the FE to be the user ID who tried to create the listing. Make sure to set the `userID` in the request body below.

```bash
curl --request POST --header 'Content-Type: application/json' -d '{"listerID": <userID>, "title": "1-BD for rent in Metrotown Station", "description": "This is going to be a long paragraph with details about the house", "housingType": "1-bedroom", "rentalPrice": 1500, "listingDate": "2023-10-22", "moveInDate": "2024-01-01", "petFriendly": "true", "status": "active", "location": {"latitude": 0, "longitude": 0}, "images": ["img1", "img2", "img3"]}' localhost:3000/api/listings
```

- `PUT /api/listings/:listingId` updates listing based on its ID, provide on the request body all changes you wish to make

```bash
curl --request PUT --header 'Content-Type: application/json' --data '{"description": "Empty", "rentalPrice":1350}' localhost:3000/api/listings/:listingId
```
- `DELETE /api/listings/:listingId` deletes listing based on its ID
```bash
curl --request DELETE --header 'Content-Type: application/json' localhost:3000/api/listings/:listingId
```

### CODE REVIEW

File: backend/src/models/listingModel.js [hunk 1]
```diff
@@ -1,14 +1,53 @@
...
+        enum: ['active', 'pending', 'inactive'], 
+        required: true, 
+        default: 'active' 
+    },
+    location: { 
+        type: {
+            latitude: {type: Number, required: true}, 
+            longitude: {type: Number, required: true}
+        }, 
+        required: true },
```
Comment (maxcao13) [inline backend/src/models/listingModel.js:49]: move the closing curly bracket down

File: backend/src/models/listingModel.js [hunk 2]
```diff
@@ -1,14 +1,53 @@
...
 const ListingSchema = new mongoose.Schema({
-    listerID: { type: mongoose.Schema.Types.ObjectId, required: true },
-    name: { type: String, required: true, minlength: 5, maxlength: 30 },
-    description: { type: String },
-    rental_price: { type: Number, required: true },
-    date: { type: Date, required: true },
-    pet_friendly: { type: Boolean },
-    active: { type: Boolean, default: true, required: true },
-    location: { type: String, coordinates: [Number], required: true },
+    listerID: {
```
Comment (maxcao13) [inline backend/src/models/listingModel.js:4]: should we just rename to userId for simplicity?

Comment (dlalaj) [reply backend/src/models/listingModel.js:4]: Yes!

File: backend/src/models/listingModel.js (hunk 1 above)
Comment (dlalaj) [reply backend/src/models/listingModel.js:49]: Sure

**Pull request merged**